src/imports.rs

```diff
@@ -0,0 +1,301 @@
+    }
+}
+
+impl From<Vec<Keypoint>> for AnnotationImportData {
+    fn from(value: Vec<Keypoint>) -> Self {
+        AnnotationImportData {
+            polygon: Some(AnnotationImportPolygon::from(value)),
+            tag: None,
+        }
+    }
+}
+
+impl AnnotationImportAnnotation {
+    /// Creates a new polygon annotation.
+    ///
+    /// This function generates an `AnnotationImportAnnotation` instance representing a polygon annotation.
+    /// It assigns a unique ID, sets the annotation data with the provided `path` representing the polygon,
+    /// and links it to an appropriate annotation class based on the `original_annotation` and the provided
+    /// `eligible_annotation_classes`. The `slot_name` is used to specify where in the dataset item this annotation
+    /// should be attached.
+    ///
+    /// # Arguments
+    ///
+    /// * `original_annotation` - A reference to an `ImageAnnotation` from which the name of the annotation class is derived.
+    /// * `path` - A vector of `Keypoint` objects defining the vertices of the polygon.
+    /// * `eligible_annotation_classes` - A slice of references to `AnnotationClass` objects.
+    ///    The function searches these to find a matching class ID for the `original_annotation`.
+    /// * `slot_name` - The name of the slot in the dataset item where this annotation will be attached.
+    ///
+    /// # Returns
+    ///
+    /// Returns `Result<AnnotationImportAnnotation, Error>` where `Ok` contains the newly created `AnnotationImportAnnotation`
+    /// if the annotation class corresponding to `original_annotation` is found in `eligible_annotation_classes`;
+    /// otherwise, an error is returned.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if no matching annotation class ID is found in `eligible_annotation_classes` for the `original_annotation`.
+    pub fn new_polygon_annotation(
+        original_annotation: &ImageAnnotation,
+        path: Vec<Keypoint>,
+        eligible_annotation_classes: &[&AnnotationClass],
+        slot_name: &str,
+    ) -> Result<Self> {
+        Ok(AnnotationImportAnnotation {
+            id: uuid::Uuid::new_v4().to_string(),
+            data: AnnotationImportData::from(path),
+            annotation_class_id: Self::find_annotation_class_id(
+                eligible_annotation_classes,
+                &original_annotation.name,
+            )?,
+            context_keys: AnnotationContext {
+                slot_names: vec![slot_name.to_string()],
+            },
+        })
+    }
+
+    /// Creates a new tag annotation.
+    ///
+    /// This function generates an `AnnotationImportAnnotation` instance for a tag annotation.
+    /// It assigns a unique ID, sets the annotation data based on the `original_annotation`,
+    /// and identifies the correct annotation class from `eligible_annotation_classes`.
+    /// The `slot_name` is used to specify the dataset item slot for the annotation.
```

Comment:
**praise**: This makes sense! Thanks for adding this - it is super clear to me 🥇